Stream-Three/Unit14-Custom_user_and_email_authentication/auth_demo/accounts/models.py

```diff
@@ -0,0 +1,35 @@
+from __future__ import unicode_literals
+
+from django.contrib.auth.models import AbstractUser, UserManager
+from django.db import models
+from django.utils import timezone
+
+# Create your models here.
+class AccountUserManager(UserManager):
+    def _create_user(self, username, email, password,
+                     is_staff, is_superuser, **extra_fields):
+        """
+       Creates and saves a User with the given username, email and password.
+       """
+        now = timezone.now()
+        if not email:
+            raise ValueError('The given username must be set')
+
+        email = self.normalize_email(email)
+        user = self.model(username=email, email=email,
+                          is_staff=is_staff, is_active=True,
+                          is_superuser=is_superuser,
+                          date_joined=now, **extra_fields)
+        user.set_password(password)
+        user.save(using=self._db)
+
+        return user
+
+
+class User(AbstractUser):
+    # now that we've abstracted this class we can add any
+    # number of custom attribute to our user class
+
+    # in later units we'll be adding things like payment details!
```

Comment:
Regarding that AccountUserManager class We don't need it.

The way the LMS explains how to do auth is wrong.
I've done a much simpler version here
https://github.com/richardadalton/follow_along_blog

I need to type up how auth works in that. And I need to do it soon because we're causing grief for online students with the lesson that's there.

-Richard